src/ec_capture.c

```diff
@@ -127,28 +141,25 @@ void capture_getifs(void)
      
       /* fill the empty descriptions */
       if (dev->description == NULL)
-         dev->description = dev->name;
-
-      /* remove the pseudo device 'any' 'nflog' and 'nfqueue' */
-      /* remove the pseudo device 'dbus-system' and 'dbus-session' shown on mac when ran without sudo*/
-      if (!strcmp(dev->name, "any") || !strcmp(dev->name, "nflog") || !strcmp(dev->name, "nfqueue")  || !strcmp(dev->name, "dbus-system") || !strcmp(dev->name, "dbus-session")) {
-         /* check if it is the first in the list */
-         if (dev == EC_GBL_PCAP->ifs)
-            EC_GBL_PCAP->ifs = ndev;
-         else
-            pdev->next = ndev;
-
-         SAFE_FREE(dev->name);
-         SAFE_FREE(dev->description);
-         SAFE_FREE(dev);
+         dev->description = strdup(dev->name);
```

Comment:
mmm where is the SAFE_FREE for this strdup?

Reply:
Good point. let me try if this is catched by `pcap_freealldevs()` or if it will result in a double-free exception.

Reply:
I've tried to address your concern in PR #1091